content/aips/942c023868ca-AIP-STKAAVE_RETRIEVAL.md

```diff
@@ -0,0 +1,77 @@
+
+[0x02d60b84491589974263d922d9cc7a3152618ef6](https://etherscan.io/address/0x02d60b84491589974263d922d9cc7a3152618ef6)
+[0xd093fa4fb80d09bb30817fdcd442d4d02ed3e5de](https://etherscan.io/address/0xd093fa4fb80d09bb30817fdcd442d4d02ed3e5de)
+[0xf8fd466f12e236f4c96f7cce6c79eadb819abf58](https://etherscan.io/address/0xf8fd466f12e236f4c96f7cce6c79eadb819abf58)
+
+Aave enables other addresses to claim on behalf of smart contracts using the `claimOnBehalfOf` function. 
+
+In order to do this, a specified address must be declared the valid owner of the contract.
+
+# Specification
+
+Set Balancer contract [`0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f`](https://etherscan.io/address/0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f) as the StkAaveRetrieval contract in the incentives controller contract, [4].
+
+Snapshot and forum discussion can be found at reference [6] and [7].
+
+# Test Cases
+
+This proposal has been tested and peer reviewed by Bored Ghost Developing, [5].
+
+Test cases, payload, and retrieval contract can be found at reference [8], [9] & [10] respectively.
+
+# Implementation
+ 
+Call setClaimer(0xba12222222228d8ba445958a75a0704d566bf2c8, RETRIEVAL_CONTRACT_ADDR) for Balancer contracts on INCENTIVES_CONTROLLER_PROXY_ADDRESS
+ 
+Balancer DAI-aDAI Linear Pool [0x02d60b84491589974263d922d9cc7a3152618ef6](https://etherscan.io/address/0x02d60b84491589974263d922d9cc7a3152618ef6)
+
+Balancer USDC-aUSDC Linear Pool [0xd093fa4fb80d09bb30817fdcd442d4d02ed3e5de](https://etherscan.io/address/0xd093fa4fb80d09bb30817fdcd442d4d02ed3e5de)
+
+Balancer USDT-aUSDT Linear Pool
+[0xf8fd466f12e236f4c96f7cce6c79eadb819abf58](https://etherscan.io/address/0xf8fd466f12e236f4c96f7cce6c79eadb819abf58)
+ 
+Incentives controller proxy [0xd784927Ff2f95ba542BfC824c8a8a98F3495f6b5[(https://etherscan.io/address/0xd784927Ff2f95ba542BfC824c8a8a98F3495f6b5#readProxyContract) 
+
+# References
+
+[1] https://docs.balancer.fi/products/balancer-pools/boosted-pools#linear-pools 
+[2] https://governance.aave.com/t/arc-continue-liquidity-mining-program-on-aave-v2-ethereum-market-and-introduce-liquidity-mining-on-aave-arc-market/7189
+[3] https://app.balancer.fi/#/pool/0x7b50775383d3d6f0215a8f290f2c9e2eebbeceb20000000000000000000000fe
+[4] https://etherscan.io/address/0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f
+[5] https://twitter.com/bgdlabs
+[6] https://snapshot.org/#/aave.eth/proposal/0xdaa660ea59f8678748d6f133d7d7ed70b941798aa9a0044a16a1285d09e26bf5
+[7] https://governance.aave.com/t/arc-whitelist-balancer-s-liquidity-mining-claim/9724
+[8] https://github.com/djh58/aave-stkaave-retrieval-llama/blob/main/src/test/StkAaveRetrievalE2E.t.sol
+[9] https://github.com/djh58/aave-stkaave-retrieval-llama/blob/main/src/ProposalPayload.sol
+[10] https://github.com/djh58/aave-stkaave-retrieval-llama/blob/main/src/StkAaveRetrieval.sol
```

Comment:
Make these into a Markdown numbered list: (i.e 1. 2. 3. instead of [1], [2], [3] etc.).

Reply:
And make all of them into Markdown links. i.e like [https://docs.balancer.fi/products/balancer-pools/boosted-pools#linear-pools](https://docs.balancer.fi/products/balancer-pools/boosted-pools#linear-pools)

Reply:
i.e using []()